src/kilo.c

```diff
@@ -318,6 +318,40 @@ void editorDelRow(int at) {
     E.dirty++;
 }
 
+void editorDeleteLines(int brow, int erow) {
+  if (brow > erow)
+    SWAP(brow, erow);
+
+  int iter = erow - brow;
+  int uppos = E.cy + E.rowoff - iter;
+  if (uppos < 0)
+    uppos = 0;
+
+  while (iter--)
+    editorDelRow(brow);
+
+  E.cy = uppos;
+}
+
+void editorDeleteSelection(int brow, int bpos, int erow, int epos) {
+  if (bpos > epos)
+    SWAP(bpos, epos);
+
+  /* Join the begin of brow and the end of erow together */
```

Comment:
If we have a character selection, we must create a new string that is the concatenation of everything before the first selection point, and everything after the second selection point. This will be the new line.